### PR TITLE
Improved UX for price-input

### PR DESCRIPTION
**Pull request by MananTank**
## Issue 

* Changing from `0.7` to `0.8` requires deleting a character and typing '8'
* when we do that - input shows `8` instead of `0.8`

Video of issue and fixed behavior:

https://user-images.githubusercontent.com/22043396/212725142-35616b1b-1c4a-4b72-a166-9f7a771234c0.mov

### CODE REVIEW

File: contract-ui/tabs/claim-conditions/components/price-input.tsx [hunk 1]
```diff
@@ -33,6 +33,8 @@ export const PriceInput: React.FC<PriceInputProps> = ({
   return (
     <InputGroup {...restInputProps}>
       <Input
+        step={0.00000000000001}
```
[the remark this answers is not in the document]
Comment (MananTank) [reply contract-ui/tabs/claim-conditions/components/price-input.tsx:36]: yeah, let me make it 18 decimals to allow for the smallest value

[the remark this answers is not in the document]
Comment (nachoiacovino) [reply contract-ui/tabs/claim-conditions/components/price-input.tsx]: right but that wouldn't always be right, if someone selects USDC, that currency only has 6 decimals. 

works for 99% of cases though, tell me if you think it's not worth, I can find the code I was telling you about.

[the remark this answers is not in the document]
Comment (MananTank) [reply contract-ui/tabs/claim-conditions/components/price-input.tsx]: That issue with adding more than 7 decimals is still there - if user enters `0.0000001` it becomes `1e-7` and we get below error 

```js
ZodError: [
  {
    "validation": "regex",
    "code": "invalid_string",
    "message": "Invalid amount",
    "path": []
  }
]
```

[the remark this answers is not in the document]
Comment (nachoiacovino) [reply contract-ui/tabs/claim-conditions/components/price-input.tsx]: great

this is the code I was telling you about: https://github.com/thirdweb-dev/dashboard/blob/main/contract-ui/tabs/tokens/components/transfer-form.tsx#L69

**Comment by MananTank** [comment]
@nachoiacovino 
Added support for prices < 10^-7 which currently does not work on prod

https://user-images.githubusercontent.com/22043396/212754609-aaf76dfa-bc44-4816-ad66-daa553278c91.mov

**Review by nachoiacovino — APPROVED** [review]
I think it's good to ship, the limiting to the amount of decimals the selected currency has is QoL so probably not worth it right now

**Pull request merged**